tests/test_varname.py

```diff
@@ -10,19 +10,6 @@
                      _get_node,
                      nameof)
 
-@pytest.fixture
-def monkey_patch():
-    """Monkey-patch inspect.stack to get only one frame,
-    which happens in R package reticulate."""
-    from varname import inspect
-
-    orig_stack = inspect.stack
-    inspect.stack = lambda *args, **kwargs: orig_stack(*args, **kwargs)[:1]
-
-    yield
-
-    inspect.stack = orig_stack
-
```

Comment:
Does this mean it will work in the environment like `reticulate` in `R` which cuts the `stacks`?

Reply:
Oooooh, I'm only understanding now what this was about. Guess I didn't fix anything.

What's the goal here? varname isn't going to work in certain environments, why half support it?

Reply:
How are you using reticulate? When I use it it doesn't cut the stack, but it also doesn't find the source code so it fails anyway:

```
> library(reticulate)
> source_python("/Users/alexhall/Library/Application Support/JetBrains/PyCharm2020.1/scratches/scratch_1005.py")
Error in py_run_file_impl(file, local, convert) : 
  IndexError: list index out of range

Detailed traceback: 
  File "<string>", line 20, in <module>
  File "<string>", line 17, in function2
  File "<string>", line 13, in function1
  File "<string>", line 9, in function
  File "/Users/alexhall/Desktop/python/python-varname/varname.py", line 121, in varname
    node = _get_node(caller)
  File "/Users/alexhall/Desktop/python/python-varname/varname.py", line 40, in _get_node
    return list(exet.statements)[0]
```

Reply:
We have another package using this one and is used in such an environment. I don't want to break that...
What is the version? Guessing mine is out-dated...
Would you post your `sessionInfo()` if you don't mind?

Reply:
But what do you do with `var_0` or whatever it returns? How is that useful?

Reply:
`var_0` was a bad idea. I guess we should raise an exception instead. That's why `raise_exc` argument was added later.

And probably we should give a `DeprecatedWarning` if people try to return `var_0`. Instead, they should catch the exception and assigned whatever fallback name they want.

Reply:
In semantic versioning, a version starting with `0.` means that the API is unstable and may change. I think that applies especially to a dark magic library which is obviously experimental. I think you can bump to 0.3.0 and skip straight to raising an exception by default, no more warnings, no more var_0. Anyone who needs the old behaviour for compatibility can stick to 0.2.0. If they want it in a future version they can request it and you can design a better API based on real use cases.

Regarding reticulate:

```
> sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS High Sierra 10.13.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] reticulate_1.16

loaded via a namespace (and not attached):
[1] compiler_4.0.2  Matrix_1.2-18   Rcpp_1.0.5      grid_4.0.2     
[5] jsonlite_1.7.0  lattice_0.20-41
> repl_python()
Python 3.6.10 (/Users/alexhall/Library/r-miniconda/envs/r-reticulate/bin/python)
Reticulate 1.16 REPL -- A Python interpreter in R.
```

Reply:
Agree with skipping to `0.3.0`. Thanks for the session info, I will check it out.

Reply:
@alexmojaki 

Not sure if this is a good place to explain the concern about `reticulate` here. I myself had a lot of code running in R but using packages in python including this one. I was wrong about the stacks. `reticulate` doesn't cut any stacks, but it fails `executing`.

If I do this:
```
try:
        frame = inspect.stack()[caller+2].frame
except IndexError:
        return None
else:
        exet = executing.Source.executing(frame)
```
Then both `exet.node` and `exet.statements` will be empty (`None` and empty `set()`, respectively); while in `python` we should be able to, at least, get the statements. Since you are also the author of `executing`, should I submit an issue there, or do you think this is the feature to support/enhance in your plan for `executing`?

Or it can be solved by #9 ?

Reply:
No, I mentioned the same thing when I tried it, see above. If `statements` is empty that probably means the source code isn't available. Then `exet.source.text` is empty too.

Reply:
Gotcha. It doesn't fix or break anything. But any chance with `bytecode` then?

Reply:
`executing` relies on the source code. Doing anything with only bytecode is hard.

Reply:
I see. Let's merge PRs now.